src/scripts/main.js

```diff
@@ -1,3 +1,44 @@
 'use strict';
 
-// write code here
+const parents = document.querySelector('ul');
+const listArr = [...document.querySelectorAll('li')];
+
+function sortList(list) {
+  const sortedList = list.sort((a, b) => {
+    return (
+      convertToNumber(b.dataset.salary) - convertToNumber(a.dataset.salary)
```

Comment:
Ensure that the `convertToNumber` function correctly handles all possible formats of the salary string. It currently assumes the string starts with a currency symbol and uses commas as thousand separators. If the format changes, this function might not work correctly.